### ISSUE TITLE
Compatibility with yarn v3

### ISSUE THREAD

**Issue by rangelfinal** [ojkelly/yarn.build#84]
Yarn v3 RC is out, and can be tested by using the command `yarn set version`  
`yarn set version 3.0.0-rc.2` for latest v3 RC  
`yarn set version from sources` for current development version  
[This issue tracks it's breaking changes](https://github.com/yarnpkg/berry/issues/1406), [and here's the changelog](https://github.com/yarnpkg/berry/blob/master/CHANGELOG.md).

Unfortunately, yarn.build seems to be incompatible with it. 

On versions `3.0.0-rc.1` and `3.0.0-rc.2`, yarn.build plugin prevents any yarn commands at all as it uses `yup`, while yarn v3 has migrated to [`typanion`](https://github.com/arcanis/typanion).

Current development version enables `yup` back to ease migration, but both `yarn build` and `yarn bundle` fail :  
  
`yarn build`
```
Type Error: e.every is not a function
    at test (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:9:12254)
    at z.validateAndExecute (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:198:525)
    at async jn.run (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:210:1845)
    at async jn.runExit (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:210:2012)
    at async i (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:302:12367)
    at async r (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:302:10721)
```
`yarn bundle`
```
Type Error: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:363:5)
    at validateString (node:internal/validators:119:11)
    at Object.join (node:path:1176:7)
    at /path/to/my/repo/.yarn/plugins/@ojkelly/plugin-build.cjs:5:1442
    at It.mktempPromise (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:178:62552)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async d.execute (/path/to/my/repo/.yarn/plugins/@ojkelly/plugin-build.cjs:5:1356)
    at async d.validateAndExecute (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:198:618)
    at async jn.run (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:210:1845)
    at async jn.runExit (/path/to/my/repo/.yarn/releases/yarn-sources.cjs:210:2012)
```

**Comment by ojkelly** [comment]
Thanks for the heads up. That shouldn't be too much work to update.

**Comment by ojkelly** [comment]
This is now fixed, and yarn 2 compatibility has been moved to `/v2`. Fixed in #88 and tested on `3.0.0-rc.2`

yarn 3
`yarn plugin import https://yarn.build/latest`

yarn 2
`yarn plugin import https://yarn.build/v2`
